p2p/server.py

```diff
@@ -264,6 +267,19 @@ def _make_peer_pool(self) -> PeerPool:
 
         if self.peer_pool.is_full:
             peer.disconnect(DisconnectReason.too_many_peers)
+        elif not self.peer_pool.is_valid_connection_candidate(peer.remote):
+            peer.disconnect(DisconnectReason.too_many_peers)
+
+        total_peers = len(self.peer_pool.connected_nodes)
+        inbound_peer_count = len([
+            peer
+            for peer
+            in self.peer_pool.connected_nodes.values()
+            if peer.inbound
+        ])
+        if total_peers > 1 and inbound_peer_count / total_peers > DIAL_IN_OUT_RATIO:
+            # make sure to have at least 1/4 outbound connections
+            peer.disconnect(DisconnectReason.useless_peer)
```

Comment:
So, `DisconnectReason.useless_peer` is really the right reason for this case?

Reply:
No, my mistake.  updated to use `too_many_peers` (couldn't find what go-ethereum does)